File: Sismique/ClientApp/app/components/shared/reportcollection.ts
import { Report, IReportProperties } from "./report";

export interface IReportCollectionProperties {
    reportId: number;
    userId: number;
    report: IReportProperties;
}

export class ReportCollection {
    reportId: number;
    userId: number;
    report: Report;

    deserialize(input: IReportCollectionProperties): ReportCollection {
        this.reportId = input.reportId;
        this.userId = input.userId;
        if (input.report != null) {
            this.report = new Report();
            this.report.deserialize(input.report);
        }
        return this;
    }
}

File: Sismique/Models/ReportCollection.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace ARIO.Models
{
    public class ReportCollection
    {
        public int ReportID { get; set; }
        public Report Report { get; set; }
        public int UserID { get; set; }
        public User User { get; set; }
    }
}